src/tsickle.ts

```diff
@@ -217,18 +222,64 @@ function addClutzAliases(
 
   // .d.ts files can be transformed, too, so we need to compare the original node below.
   const origSourceFile = ts.getOriginalNode(sourceFile);
-  // The module exports might be re-exports, and in the case of "export *" might not even be
-  // available in the module scope, which makes them difficult to export. Avoid the problem by
-  // filtering out symbols who do not have a declaration in the local module.
+  // In order to write aliases, the exported symbols need to be available in the
+  // the module scope. That is not always the case:
+  //
+  // 1) export const X;           // works
+  // 2) export {X} from './foo';  // doesn't
+  // 3) import {X} from './foo';  // works
+  //    export {X} from './foo';
+  //
+  // getExportsOfModule returns all three types, but we need to separate 2).
+  // For now we 'fix' 2) by simply not emitting a clutz alias, since clutz
+  // interop is used in minority of scenarios.
+  //
+  // TODO(radokirov): attempt to add appropriate imports for 2) so that
+  // currently finding out local appears even harder than fixing exports.
   const localExports = moduleExports.filter(e => {
-    // If there are no declarations, be conservative and emit the aliases.
-    if (!e.declarations) return true;
+    // If there are no declarations, be conservative and don't emit the aliases.
+    // I don't know how can this happen, we have no tests that excercise it.
+    if (!e.declarations) return false;
+
     // Skip default exports, they are not currently supported.
-    // default is a keyword in typescript, so the name of the export being default means that it's a
-    // default export.
+    // default is a keyword in typescript, so the name of the export being
+    // default means that it's a default export.
     if (e.name === 'default') return false;
-    // Otherwise check that some declaration is from the local module.
-    return e.declarations.some(d => d.getSourceFile() === origSourceFile);
+
+    // Use the declaration location to determine separate cases - 1), 2) and 3).
+    for (const d of e.declarations) {
+      // This is a special case for export *. Technically, it is neither case 1)
+      // 2) or 3) originally, but at this point we have rewritten it to 2) or 3)
+      // However, it appears that the rewriting also has made it behave
+      // different from explicit named export in the sense that the declaration
+      // appears to point at the original location not the reexport location.
+      // Since we can't separate 2) from 3) here, we err on the side of less
+      // emit.
+      if (d.getSourceFile() !== origSourceFile) {
+        return false;
+      }
+
+      if (!ts.isExportSpecifier(d)) {
+        // we are in case 1) thus safe to emit clutz alias.
+        return true;
+      }
+
+      // Original declarations are useless to separate 2) and 3) because they
```

Comment:
not sure what original means here?

Reply:
e.declarations in the current context. Reworded.